Lib/zipfile.py

```diff
@@ -1250,14 +1250,43 @@ def infolist(self):
         archive."""
         return self.filelist
 
-    def printdir(self, file=None):
-        """Print a table of contents for the zip file."""
-        print("%-46s %19s %12s" % ("File Name", "Modified    ", "Size"),
+    def printdir(self, file=None, verbose=1):  # N.B. Verbose 1 is as original
+        """
+        Print a table of contents for the zip file.
+        :param file: Output file descriptor
+        :param verbose: How much detail to output"
+        """
+        formatinfo = [
+            # (Title, Width, Fmt) In order of verbosity
+            ("File Name", 46, "{z.filename:46s}"),  # verbose = 0
+            ("Modified    ", 23,
+             "{z.date_time[0]:4d}-{z.date_time[1]:02d}-{z.date_time[2]:02d} "
+             "{z.date_time[3]:02d}:{z.date_time[4]:02d}:{z.date_time[5]:02d}"),  # verbose = 1
+            ("Size", 12, "{z.file_size:>12d}"),  # verbose = 1
+            ("Comp", 12, "{z.compress_size:>12d}"),  # verbose = 1
+            ("%", 7, "{percent:>8}"),   # verbose = 2
+            ("CRC", 12, "{z.CRC:12}"),  # verbose = 3
+            ("Method", 11, "{compression:>11}"),  # verbose = 3
+        ]
+        elsement_dict = {0:1, 1:3, 2:4, 3:5, 4:6,}
+        fieldcount = elsement_dict.get(verbose, len(formatinfo))
```

Comment:
So if verbose is out of the known range, print everything? You might make a comment to that effect.